psakefile.ps1

```diff
@@ -0,0 +1,54 @@
+include "./psake-build-helpers.ps1"
+
+$tag = $(git tag -l --points-at HEAD)
+$revision = @{ $true = "{0:00000}" -f [convert]::ToInt32("0" + $env:BUILD_BUILDID, 10); $false = "local" }[$env:Build.BuildId -ne $NULL];
+$suffix = @{ $true = ""; $false = "ci-$revision"}[$tag -ne $NULL -and $revision -ne "local"]
+
+properties {
+	$configuration = 'Release'
+	$projectRoot = Resolve-Path "./"
+	$buildDir = "$projectRoot/Build"
+	$testResultsDir = "$projectRoot/src/BulkWriter.Tests/TestResults"
+	$versionPrefix = "3.0.0"
+	$version = "ci-local"
+}
+
+task default -depends Run-Tests
+task Run-CI -depends Clean-Solution, Run-Tests -description "Continuous Integration process"
+task Run-LocalCI -depends Clean-Solution, Run-Tests, Package-Local -description "Local continuous integration process"
+task ReBuild-Solution -depends Clean-Solution, Run-Tests -description "Rebuild the code, with testing"
+
+task Get-Info -description "Display runtime information" {
+	exec { dotnet --info }
+}
+
+task Run-Tests -depends Run-Restore, Build-Solution -description "Run unit tests" {
+	exec { dotnet test --configuration $configuration  --logger "trx;LogFileName=Testresults.xml" } -workingDirectory src/BulkWriter.Tests
+}
+
+task Build-Solution -depends Get-Info -description "Build the solution" {
+	if($suffix -eq "") {
+		set-project-properties $versionPrefix
+	}
+	else
+	{
+	    set-project-properties $version
+	}
+	exec { dotnet build BulkWriter.sln -c $configuration -v q /nologo }
+}
+
+task Clean-Solution -description "Clean out all the binary folders" {
+	exec { dotnet clean --configuration $configuration /nologo } -workingDirectory $projectRoot
+	remove-directory-silently $testResultsDir
+	remove-directory-silently $buildDir
+}
+
+task Run-Restore -depends Get-Info -description "Restore dependencies and tools" {
+	exec { dotnet restore }
```

Comment:
This will not restore tools, you need to use `dotnet tool restore` for that instead.
Also, `dotnet build` will do restore automatically, so you don't need to do it explicitly. You can add `--no-restore` to skip it if you want a separate restore step.

Reply:
The setup script does the tool restore and the build is set to --no-restore.